### ISSUE TITLE
Facebook authentication redirect broken for WebGL games due to iframe embed

### ISSUE THREAD

**Issue by dmarchand** [itchio/itch.io#659]
WebGL games are rendered in an iframe, which points to the CDN.

Facebook auth redirect requires all requests to originate from a whitelisted domain, which also must match the website URL.

You can use the CDN as the primary domain for your app (v6p9d9t4.ssl.hwcdn.net), however this causes issues with the FB authentication popup for some users.

A similar issue also causes Unity's persistence data (PlayerPrefs) to be lost on WebGL app rebuilds.

**Comment by leafo** [comment]
Do you know what issue the CDN domain is causing? Is it because the redirect needs to go back to the itch.io main page? Maybe if itch.io provided a top level api that proxied a facebook login request?

**Comment by dmarchand** [comment]
Yeah, that's the issue. Another solution would be for itch to use a reverse proxy, so content stored on the CDN appears to be coming from itch.io. This would solve both the PlayerPrefs issue and the FB issue, I believe.

**Comment by pgregory** [comment]
Has there been any progress towards a solution for this? I need this to be able to launch my Web game in itch.io, but users globally don't seem to be able to use the Facebook login button. I've tested it here with the cdn approach, and it works, I can only assume that the cdn address might change based on geographic location, as the first failure that was reported by a beta tester was in Chile.

**Comment by fasterthanlime** [comment]
Just a few thoughts:

  * Proxying all game data would defeat the purpose of having a CDN (since all traffic would be going through our servers again instead of being served directly from edge nodes distributed around the globe)
  * Serving game content directly from `itch.io` would be a security issue - that's the reason behind the `github.com` / `github.io` split, for example - but we could (afaict) add `itch.zone` as a domain to our CDN configuration so it's at least predictable

(I'm not sure how to solve the issue at hand yet, but these seemed like important points to make)

**Comment by leafo** [comment]
The current CDN URL will not change for the time being, but we made the mistake of not providing a custom domain with our current provider when we initially set it up. That means if we ever change providers the domain will change have to change. If that happens we'll give as much notice as we can. If we do change CDNs then the domain will change to a `itch.zone` subdomain. If you want to white-list domains, then you should add the current one `v6p9d9t4.ssl.hwcdn.net` and if possible, a wildcard for `*.itch.zone`.
